src/pages/Admin.js

```diff
@@ -34,37 +36,321 @@ const RiseUp = styled.div`
-  flex-direction: column;
-  width: 400px;
+  flex-direction: row;
+  padding-bottom: 40px;
+`;
+
+const SectionTitle = styled.p`
+  font-size: 22px;
+  color: ${theme.text.darker_default};
+  line-height: normal;
+  font-weight: 500;
+`;
+
+const SectionText = styled.p`
+  text-align: left;
+  line-height: 30px;
+  margin-top: 5px;
+  font-size: 17px;
+  color: #546978;
+  margin-bottom: 20px;
 `;
 
-const ProposalPanel = styled(Card)`
-  max-width: 650px;
-  padding: 0px 25px 18px 25px;
+const StyledBody = styled.p`
+  width: 150px;
+  text-align: left;
+  line-height: 30px;
+  margin-top: 5px;
+  font-size: 17px;
   color: #546978;
+`;
+
+const OptionText = styled.p`
+  text-align: left;
   line-height: 30px;
+  font-size: 17px;
+  color: #546978;
+`;
+
+const VoteOptionsGrid = styled.div`
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-column-gap: 10px;
+  grid-row-gap: 10px;
+`;
+
+const CloseIcon = styled.p`
+  width: 15px;
+  height: 15px;
+  background-color: red;
+  mask: url(${closeImg}) center no-repeat;
+  cursor: pointer;
+`;
+
+const Code = styled.pre`
+  font-size: 14px;
+  padding: 30px;
 `;
 
+const ABSTAIN = 'Abstain';
+const NOCHANGE = 'No Change';
+
 class Admin extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      pollTitle: '',
+      pollSummary: '',
+      pollStart: null,
+      pollEnd: null,
+      pollLink: '',
+      pollOption: '',
+      pollOptions: [ABSTAIN, NOCHANGE],
+      pollContent: '',
+      pollMarkdown: '',
+      pollGenerated: false
+    };
+  }
+
+  handlePollState = (e, k) => {
+    e.stopPropagation();
+    this.setState({
+      ...this.state,
+      [k]: e.target.value
+    });
+  };
+
+  addPollVoteOption = () => {
+    if (this.state.pollOption.length) {
+      const opt = this.state.pollOption;
+      const opts = this.state.pollOptions;
+      this.setState({
+        pollOption: '',
+        pollOptions: [...opts, opt]
+      });
+    }
+  };
+
+  removePollOption = idx => {
+    const opts = this.state.pollOptions.filter((item, index) => index !== idx);
+    this.setState({
+      pollOptions: opts
+    });
+  };
+
+  resetPollState = () => {
+    this.setState({
+      pollTitle: '',
+      pollSummary: '',
+      pollLink: '',
+      pollOption: '',
+      pollOptions: [ABSTAIN, NOCHANGE],
+      pollContent: ''
+    });
+  };
+
+  parseFormToMarkdownString = () => {
+    const optsString = this.state.pollOptions.reduce(
+      (acc, opt, idx) => `${acc}   ${idx}: ${opt}\n`,
+      ''
+    );
+    const yml = `---\ntitle: ${this.state.pollTitle}\nsummary: ${
+      this.state.pollSummary
+    }\ndiscussion_link: ${this.state.pollLink}\noptions:\n${optsString}---\n`;
+    const md = `# Poll: ${this.state.pollTitle}\n\n${this.state.pollContent}`;
+    this.setState({
+      pollMarkdown: `${yml}${md}`,
+      pollGenerated: true
+    });
+  };
+
   render = () => {
+    const isValidSubmission =
+      !!this.state.pollTitle &&
```

Comment:
for neatness it might be nice to destructure some of the state variables at the top of the render fnc